internal/executor/job/processors/remove_runs.go

```diff
@@ -0,0 +1,69 @@
+package processors
+
+import (
```

Comment:
My OO brain says that the terminal event processors are 90% the same and we could therefore exrtract some of this out.